docker/tensorflow-application/analysis/analysis.py

```diff
@@ -0,0 +1,209 @@
+def show_inference(input_images, output_dict):
+    """ Decodes JSON data and converts it to a bounding box overlay
+            on the input image, then saves the image to a directory.
+
+        Args:
+            input_image: The string representing the input image.
+            response: The list of response dictionaries from the server.
+            i: An integer used in iteration over input images.
+    """
+    out_images = []
+    for num, image in enumerate(input_images):
+        output_dict_tmp = output_dict[num]
+        print(image.shape)
+        print(output_dict_tmp)
+        vis_util.visualize_boxes_and_labels_on_image_array(
+            image,
+            output_dict_tmp['detection_boxes'],
+            output_dict_tmp['detection_classes'],
+            output_dict_tmp['detection_scores'],
+            category_index,
+            instance_masks=output_dict_tmp.get('detection_masks_reframed', None),
+            use_normalized_coordinates=True,
+            line_thickness=2,
+            min_score_thresh=.3)
+
+        out_images.append(image)
+    return(out_images)
+    #visualization_utils.save_image_array_as_png(image, output_file)
+
+#Function to read in frames and do analysis as long as video sream is active 
+def generate():
+
+    os.environ["OPENCV_FFMPEG_CAPTURE_OPTIONS"] = "protocol_whitelist;file,tcp,https,tls"
+    cap = cv2.VideoCapture("carVideo.mp4")
+
+    #Frame count 
+    i = 0
+    #Segment count
+    j = 0
+
+    #While incoming video is received from IoT Device Simulator
+    batch = []
+    while(cap.isOpened()): 
+
+        # Capture frames 
+        ret, tempframe = cap.read()
+        if ret == True: 
+            #Run inference on video Frame    
+            #Add add every other frame to inference batch for speed up)
+            if(i != 0 and i%2 == 0):
+                batch.append(tempframe)
+            #Every 5 Frames Send batch to TF_serving container 
+            if (i%10 == 0 and i != 0):
+
+                #Considering a Video with 30fps make 3 second clip with at 90fps
+                if(i == 90):
+                    #Re-Zero frame count
+                    i=0
+                    j = j+1
+
+                #Otherwise send frames for inference and append them to the current video segment 
+                else:
+                    out_batch = run_inference_for_batch_of_images(tf_url,batch)
+                    #Send out_batch to the video server
+                    send_to_videoserver(out_batch)
+                batch = []
+
+                i = i + 1
+            else:
+                #Be sure to always increment frame number
+                i = i + 1
+                continue
+    #Clear variables before closing module 
+    j = 0
+    i = 0
+
+def send_to_videoserver(images):
+	payload = {"instances": [np.array(image).tolist() for image in images]}
+    # TODO Change to streaming service's IP
+	response = requests.post("http://0.0.0.0:8080/video_stream", json=payload)
```

Comment:
I don't know if you're planning on tackling this here or in the future, but a good way to make this configurable is to use an environment variable to specify the base URL.

Reply:
I was not planning on implementing that at the moment, I can look into that in a following PR if that's okay with you.

Reply:
Yeah, future changes are fine